From 86b2e78253870fd335156fc32db0d3e3e8f28599 Mon Sep 17 00:00:00 2001
From: Christian Bruel <christian.bruel@foss.st.com>
Date: Tue, 29 Jul 2025 10:11:28 +0200
Subject: [PATCH] misc: pci_endpoint_test: Add deviceID for STMICRO

Add device ID specific to STMICRO in pci_endpoint_test so that
endpoints configured with those deviceIDs can use pci_endpoint_test
driver.

Set outbound memory alignment to 64K

Signed-off-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: I29c2a955c118384f74edc781d7e550587497577f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/478579
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 drivers/misc/pci_endpoint_test.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/drivers/misc/pci_endpoint_test.c
+++ b/drivers/misc/pci_endpoint_test.c
@@ -84,6 +84,8 @@
 #define PCI_DEVICE_ID_RENESAS_R8A774E1		0x0025
 #define PCI_DEVICE_ID_RENESAS_R8A779F0		0x0031
 
+#define PCI_DEVICE_ID_STMICRO_TEST              0xCCFF
+
 static DEFINE_IDA(pci_endpoint_test_ida);
 
 #define to_endpoint_test(priv) container_of((priv), struct pci_endpoint_test, \
@@ -974,6 +976,12 @@ static const struct pci_endpoint_test_da
 	.irq_type = IRQ_TYPE_MSI,
 };
 
+static const struct pci_endpoint_test_data stm32mp2_data = {
+	.test_reg_bar = BAR_0,
+	.alignment = SZ_64K,
+	.irq_type = IRQ_TYPE_MSI,
+};
+
 static const struct pci_device_id pci_endpoint_test_tbl[] = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_TI, PCI_DEVICE_ID_TI_DRA74x),
 	  .driver_data = (kernel_ulong_t)&default_data,
@@ -1011,6 +1019,9 @@ static const struct pci_device_id pci_en
 	{ PCI_DEVICE(PCI_VENDOR_ID_TI, PCI_DEVICE_ID_TI_J721S2),
 	  .driver_data = (kernel_ulong_t)&j721e_data,
 	},
+	{ PCI_DEVICE(PCI_VENDOR_ID_STMICRO, PCI_DEVICE_ID_STMICRO_TEST),
+	  .driver_data = (kernel_ulong_t)&stm32mp2_data,
+	},
 	{ }
 };
 MODULE_DEVICE_TABLE(pci, pci_endpoint_test_tbl);
